packages/frontend/src/app/modules/flow-builder/page/flow-builder/flow-right-sidebar/edit-step-sidebar/edit-step-accordion/input-forms/piece-input-forms/piece-action-input-form/piece-action-input-form.component.ts

```diff
@@ -151,26 +153,22 @@ export class PieceActionInputFormComponent
     );
   }
 
-  fetchActions(pieceName: string) {
-    const pieces$ = this.actionMetaDataService.getPieces().pipe(
-      map((pieces) => {
-        const component = pieces.find((c) => c.name === pieceName);
-        if (!component) {
-          throw new Error(`Activepieces- piece not found: ${pieceName}`);
-        }
-        return component;
-      })
+  fetchActions(pieceName: string, pieceVersion: string) {
+    const pieceMetadata$ = this.actionMetaDataService.getPieceMetadata(
+      pieceName,
+      pieceVersion
     );
-    this.actions$ = pieces$.pipe(
-      map((component) => {
-        const actionsKeys = Object.keys(component.actions);
+
+    this.actions$ = pieceMetadata$.pipe(
+      map((pieceMetadata) => {
+        const actionsKeys = Object.keys(pieceMetadata.actions);
         return actionsKeys.map((actionName) => {
-          const action = component.actions[actionName];
+          const action = pieceMetadata.actions[actionName];
 
           const configs = Object.entries(action.props).map((keyEntry) => {
```

Comment:
can you please make keyEntry => [propName,prop] ? sorry I should have done that at the start.

Reply:
fixed in 6ce6500